src/main/appender.cpp

```diff
@@ -410,6 +411,31 @@ void Appender::FlushInternal(ColumnDataCollection &collection) {
 	context->Append(*description, collection);
 }
 
+void Appender::AppendDefaultToVector(Vector &result, idx_t column, SelectionVector &sel, idx_t count) {
+	// TODO: throw if count is > STANDARD_VECTOR_SIZE
+	if (column >= types.size()) {
+		throw InvalidInputException(
+		    "Could not append DEFAULT value of column %d, out of range index, table only has %d columns", column,
+		    types.size());
+	}
+	auto &type = types[column];
+	if (result.GetType() != type) {
+		throw InvalidInputException("Type mismatch, column has type: %s, while the provided Vector has type %s",
+		                            type.ToString(), result.GetType().ToString());
+	}
+
+	auto &executor = *expression_executor;
+
+	// This is only used to tell the ExpressionExecutor the size of our result
+	DataChunk empty_chunk;
+	empty_chunk.SetCardinality(count);
+	executor.SetChunk(empty_chunk);
+
+	// FIXME: if we are feeding the Vector to the ExecuteExpression method, we have to first make sure
```

Comment:
Since the `result` is arbitrary provided by the user, we should probably make an intermediate result instead and use SetValue GetValue to transfer the data between them (we can't use Reference/Slice because we don't want to disturb the existing data)

Reply:
Yes, this sounds good. I was already surprised by how you used the expression executor, as hinted at by my comment [here](https://github.com/duckdb/duckdb/pull/11905#discussion_r1593546803). But didn't have a suggestion for a suitable alternative, haha.